components/places/src/bookmark_sync/record.rs

```diff
@@ -317,6 +322,48 @@ pub enum BookmarkItemRecord {
     Separator(SeparatorRecord),
 }
 
+// dateAdded on a bookmark might be a string! See #1148.
+fn de_maybe_stringified_timestamp<'de, D>(
+    deserializer: D,
+) -> std::result::Result<Option<i64>, D::Error>
+where
+    D: serde::de::Deserializer<'de>,
+{
+    use std::fmt;
+    use std::marker::PhantomData;
+
+    struct StringOrInt(PhantomData<Option<i64>>);
```

Comment:
What does `PhantomData` do here?

Reply:
It apparently allows the type to act like it holds an `Option<i64>` without actually holding one, and TBH, I copy-pasta'd this from other serde examples trying to do similar things.

If I remove that and change the deserialize call to `deserializer.deserialize_any(StringOrInt(Some(0)))` things work, which is both somewhat confusing and smelly. 🤷‍♂

Reply:
PhantomData only would matter here if it were generic, or if Option<i64> implemented Drop, held onto references, or a couple other cases.

It doesn't really matter, though.